lib/segment/src/index/field_index/full_text_index/posting_list.rs

```diff
@@ -37,23 +40,309 @@ impl PostingList {
+            data: data.into_boxed_slice(),
+            chunks: chunks.into_boxed_slice(),
+        }
+    }
+
+    pub fn contains(&self, val: &PointOffsetType) -> bool {
+        if !self.is_in_postings_range(*val) {
+            return false;
+        }
+
+        // find the chunk that may contain the value and check if the value is in the chunk
+        let chunk_index = self.find_chunk(val, None);
+        if let Some(chunk_index) = chunk_index {
+            if self.chunks[chunk_index].initial == *val {
+                return true;
+            }
+
+            let mut decompressed = [0u32; BitPackerImpl::BLOCK_LEN];
+            self.decompress_chunk(&BitPackerImpl::new(), chunk_index, &mut decompressed);
+            decompressed.binary_search(val).is_ok()
+        } else {
+            false
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = PointOffsetType> + '_ {
+        let bitpacker = BitPackerImpl::new();
+        (0..self.chunks.len())
+            .flat_map(move |chunk_index| {
+                let mut decompressed = [0u32; BitPackerImpl::BLOCK_LEN];
+                self.decompress_chunk(&bitpacker, chunk_index, &mut decompressed);
+                decompressed.into_iter()
+            })
+            .take(self.len)
+    }
+
+    fn get_chunk_size(chunks: &[CompressedPostingChunk], data: &[u8], chunk_index: usize) -> usize {
+        assert!(chunk_index < chunks.len());
+        if chunk_index + 1 < chunks.len() {
+            chunks[chunk_index + 1].offset as usize - chunks[chunk_index].offset as usize
+        } else {
+            data.len() - chunks[chunk_index].offset as usize
+        }
+    }
+
+    fn find_chunk(&self, doc_id: &PointOffsetType, start_chunk: Option<usize>) -> Option<usize> {
+        let start_chunk = if let Some(idx) = start_chunk { idx } else { 0 };
+        match self.chunks[start_chunk..].binary_search_by(|chunk| chunk.initial.cmp(doc_id)) {
+            // doc_id is the initial value of the chunk with index idx
+            Ok(idx) => Some(start_chunk + idx),
+            // chunk idx has larger initial value than doc_id
+            // so we need the previous chunk
+            Err(idx) => {
+                if idx > 0 {
+                    Some(start_chunk + idx - 1)
+                } else {
+                    None
+                }
+            }
+        }
+    }
+
+    fn is_in_postings_range(&self, val: PointOffsetType) -> bool {
+        self.chunks.len() > 0 && val >= self.chunks[0].initial && val <= self.last_doc_id
+    }
+
+    fn decompress_chunk(
+        &self,
+        bitpacker: &BitPackerImpl,
+        chunk_index: usize,
+        decompressed: &mut [PointOffsetType],
+    ) {
+        let chunk = &self.chunks[chunk_index];
+        let chunk_size = Self::get_chunk_size(&self.chunks, &self.data, chunk_index);
+        let chunk_bits = (chunk_size * 8) / BitPackerImpl::BLOCK_LEN;
```

Comment:
Are we sure we never have any remainder here?

Reply:
We are sure about it because `chunk_size` is defined as `chunk_bits * BitPackerImpl::BLOCK_LEN / 8`, `BitPackerImpl::BLOCK_LEN` is `128`